personal_graph/graph.py

```diff
@@ -5,77 +5,1359 @@
 from __future__ import annotations
 
 import json
-from typing import Any, List, Optional, Union, Dict
+import os
+import pathlib
+import uuid
+from dataclasses import dataclass
+from functools import lru_cache
+from typing import Any, List, Optional, Union, Dict, Tuple, Callable
+
+import instructor
 import libsql_experimental as libsql  # type: ignore
 from contextlib import AbstractContextManager
+
+import networkx as nx  # type: ignore
 from graphviz import Digraph  # type: ignore
-from .models import Node, EdgeInput, KnowledgeGraph
-from .database import (
-    atomic,
-    connect_nodes,
-    initialize,
-    add_nodes,
-    upsert_node,
-    connect_many_nodes,
-    add_node,
-    remove_node,
-    remove_nodes,
-    find_node,
-    find_neighbors,
-    find_outbound_neighbors,
-    traverse,
-    pruning,
-    find_similar_nodes,
-    nodes_list,
-    vector_search_node,
-    find_label,
-    find_outdegree_edges,
-    find_indegree_edges,
-)
-from .natural import insert_into_graph, search_from_graph, visualize_knowledge_graph
-from .visualizers import graphviz_visualize
+from jinja2 import Environment, BaseLoader, select_autoescape
+from litellm.llms import openai  # type: ignore
+from matplotlib import pyplot as plt
+from dotenv import load_dotenv
+
+from personal_graph.embeddings import OpenAIEmbeddingsModel
+from personal_graph.models import Node, EdgeInput, KnowledgeGraph, Edge
+from personal_graph.visualizers import _as_dot_node, _as_dot_label
+
+load_dotenv()
+
+CursorExecFunction = Callable[[libsql.Cursor, libsql.Connection], Any]
+
+
+@dataclass
+class OpenAIClient:
+    client: openai.OpenAI = openai.OpenAI(
+        api_key="",
+        base_url=os.getenv("LITE_LLM_BASE_URL", ""),
+        default_headers={"Authorization": f"Bearer {os.getenv('LITE_LLM_TOKEN', '')}"},
+    )
+
+
+@dataclass
+class EmbeddingClient(OpenAIClient):
+    model_name: str = "openai/text-embedding-3-small"
+    dimensions: int = 384
+
+
+@dataclass
+class LLMClient(OpenAIClient):
+    model_name: str = "openai/gpt-3.5-turbo"
+
+
+@lru_cache(maxsize=None)
+def read_sql(sql_file: str) -> str:
+    with open(pathlib.Path(__file__).parent.resolve() / "sql" / sql_file) as f:
```

Comment:
this would lead to strange behaviour as __file__ could be in a seperate location than the script using this library